RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -0,0 +1,185 @@
+    var configuration: CustomerCenterConfigData?
+    @Published
+    var showRestoreAlert: Bool = false
+    @Published
+    var state: CustomerCenterViewState {
+        didSet {
+            if case let .error(stateError) = state {
+                self.error = stateError
+            }
+        }
+    }
+
+    var isLoaded: Bool {
+        return state != .notLoaded
+    }
+
+    private var purchasesProvider: ManageSubscriptionsPurchaseType
+
+    private var error: Error?
+
+    convenience init() {
+        self.init(purchasesProvider: ManageSubscriptionPurchases())
+    }
+
+    // @PublicForExternalTesting
+    init(purchasesProvider: ManageSubscriptionsPurchaseType) {
+        self.state = .notLoaded
+        self.purchasesProvider = purchasesProvider
+    }
+
+    // @PublicForExternalTesting
+    init(configuration: CustomerCenterConfigData,
+         subscriptionInformation: SubscriptionInformation) {
+        self.configuration = configuration
+        self.subscriptionInformation = subscriptionInformation
+        self.purchasesProvider = ManageSubscriptionPurchases()
+        state = .success
+    }
+
+    func loadScreen() async {
+        do {
+            try await loadSubscriptionInformation()
+            await loadCustomerCenterConfig()
+            self.state = .success
+        } catch {
+            self.state = .error(error)
+        }
+    }
+
+    func loadSubscriptionInformation() async throws {
+        let customerInfo = try await purchasesProvider.customerInfo()
+        guard let currentEntitlementDict = customerInfo.entitlements.active.first,
+              let subscribedProductID = customerInfo.activeSubscriptions.first,
+              let subscribedProduct = await purchasesProvider.products([subscribedProductID]).first else {
+            Logger.warning(Strings.could_not_find_subscription_information)
+            throw CustomerCenterError.couldNotFindSubscriptionInformation
+        }
+        let currentEntitlement = currentEntitlementDict.value
+
+        // swiftlint:disable:next todo
+        // TODO: support non-consumables
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .medium
+        self.subscriptionInformation = SubscriptionInformation(
+            title: subscribedProduct.localizedTitle,
+            durationTitle: subscribedProduct.subscriptionPeriod?.durationTitle ?? "",
+            price: subscribedProduct.localizedPriceString,
+            nextRenewalString: currentEntitlement.expirationDate.map { dateFormatter.string(from: $0) } ?? nil,
+            willRenew: currentEntitlement.willRenew,
+            productIdentifier: subscribedProductID,
+            active: currentEntitlement.isActive
+        )
+    }
+
+    func loadCustomerCenterConfig() async {
+        self.configuration = CustomerCenterConfigTestData.customerCenterData
+    }
+
+    #if os(iOS) || targetEnvironment(macCatalyst)
+    func handleAction(for path: CustomerCenterConfigData.HelpPath) {
```

Comment:
This method looks like it should be `async`, and use the `Task { }` blocks? This way the caller can present a spinner or some other "processing" UI if needed. Then to conveniently call it from a `Button`, you can use our custom `AsyncButton`:
```swift
                    AsyncButton {
                        await self.viewModel.handleAction(for: path)
                    } label: {
                        Text(path.title)
                    }
```
Disclaimer: have not tried running this code.